js/Config.js

```diff
@@ -0,0 +1,74 @@
+function FC() {
+	this._subConfigCategories = [];
+	
+	this.setConfigProvider = function(configProvider) {
```

Comment:
You should add methods to the prototype instead of to the object, unless they are instance specific.